### PR TITLE
WINC-823: Test generated community manifests in WMCO e2e Followup

### PR DESCRIPTION
**Pull request by alinaryan**
Adds a check for the COMMUNITY env var. If present(true), the manifest
location is set to the artifact dir. Otherwise, it is set to the local bundle/ location.

Moves ARTIFACT_DIR check up so that nounset is not triggered. 

Adjusts generate.sh to copy over wmco-package.yaml
to the community bundle as it's needed for e2e testing.

Depends on: https://github.com/openshift/windows-machine-config-operator/pull/1151 and https://github.com/openshift/release/pull/30711

### CODE REVIEW

File: hack/common.sh [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #!/bin/bash
 
 # Location of the manifests file
-MANIFEST_LOC=bundle/
+if [ "$COMMUNITY" = "true" ]; then
```
Comment (sebsoto) [inline hack/common.sh:4]: CI is failing because this is triggering `set -o nounset`
Try doing `if [ -n "$COMMUNITY"  && "$COMMUNITY" = "true" ]; then`

Comment (alinaryan) [reply hack/common.sh]: Thanks, I was just about to ask about this

File: hack/common.sh [hunk 2]
```diff
@@ -67,7 +71,7 @@ OSDK_WMCO_management() {
 
 build_WMCO() {
   local OSDK=$1
-  
+
```
Comment (sebsoto) [inline hack/common.sh:74]: unrelated change

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: hack/common.sh [hunk 3]
```diff
@@ -1,7 +1,11 @@
 #!/bin/bash
 
-# Location of the manifests file
-MANIFEST_LOC=bundle/
+# Set location of the manifests file
+if [ -z ${COMMUNITY:-} ]; then
+  MANIFEST_LOC=bundle/
+else
+  MANIFEST_LOC="$ARTIFACT_DIR"
+fi
```
Comment (jrvaldes) [inline hack/common.sh:8]: I would keep a default value and update as necessary. That is:
```shell
# set default location of the manifests file
MANIFEST_LOC="$ARTIFACT_DIR"
# update for commutiy 
if [ -z ${COMMUNITY:-} ]; then
  MANIFEST_LOC=bundle/ 
fi
```

**Review by jrvaldes — COMMENTED** [review]
LGTM, just minor comments.

File: hack/common.sh (hunk 3 above)
Comment (alinaryan) [reply hack/common.sh:8]: @jrvaldes Thanks, but I might keep what I have. ARTIFACT_DIR is not the default location. That if statement checks if COMMUNITY is unset and keeps the default bundle/ location without triggering `set -o nounset`

Comment (saifshaikh48) [inline hack/common.sh:8]: +1 to Jose's code structure suggestion

note that `-z` checks if a string is null (that is, has zero length) whereas `-n` checks that a string is _not_ null

```suggestion
# set default location of the manifests file
MANIFEST_LOC="$ARTIFACT_DIR"
# update for community 
if [ -n ${COMMUNITY:-} && "$COMMUNITY" = "true" ]; then
  MANIFEST_LOC=bundle/ 
fi
```

Comment (sebsoto) [reply hack/common.sh]: How is `-n ${COMMUNITY:-}` ever going to evaluate to false?
`${COMMUNITY:-}` by definition cannot be unset by

Comment (saifshaikh48) [reply hack/common.sh]: @sebsoto `-n ${COMMUNITY:-}` would evaluate to false if `COMMUNITY=""` (empty string) or the variable is not set (tested this by running `unset COMMUNITY` before the conditional). Since we default to `COMMUNITY="false"` in CI config it won't ever be false but it the check doesn't hurt imo.

Comment (sebsoto) [reply hack/common.sh]: Got it. Needs quotes around `${COMMUNITY:-}` though.
Instead of putting it into the conditional,  i suggest doing `COMMUNITY=${COMMUNITY:-false}

Comment (sebsoto) [reply hack/common.sh]: so overall the check looks like this: 
```
MANIFEST_LOC=bundle/
COMMUNITY=${COMMUNITY:-false}
if [ "$COMMUNITY" = "true" ]; then
  MANIFEST_LOC="$ARTIFACT_DIR"
fi
```

File: hack/common.sh [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 # Location of the manifests file
 MANIFEST_LOC=bundle/
+COMMUNITY=${COMMUNITY:-false}
```
Comment (jrvaldes) [inline hack/common.sh:5]: I suggest moving this to the top of the file and adding a comment that explains why the default value is set to `false`. Then, handle the MANIFEST_LOC logic.

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Comment by jrvaldes** [comment]
@alinaryan   any backports?

**Comment by jrvaldes** [comment]
> @alinaryan: #1155 failed to apply on top of branch "release-4.11":
> 
> ```
> Applying: check for COMMUNITY env var
> Using index info to reconstruct a base tree...
> M	hack/community/generate.sh
> M	hack/run-ci-e2e-test.sh
> Falling back to patching base and 3-way merge...
> Auto-merging hack/run-ci-e2e-test.sh
> Auto-merging hack/community/generate.sh
> CONFLICT (content): Merge conflict in hack/community/generate.sh
> error: Failed to merge in the changes.
> hint: Use 'git am --show-current-patch=diff' to see the failed patch
> Patch failed at 0001 check for COMMUNITY env var
> When you have resolved this problem, run "git am --continue".
> If you prefer to skip this patch, run "git am --skip" instead.
> To restore the original branch and stop patching, run "git am --abort".
> ```

Depends on:
- https://github.com/openshift/windows-machine-config-operator/pull/1239
